File: src/lib/types/item.ts
export type ItemSourceInfo = {
    id: string;
    name: string;
    item_count: number;
};

export type ItemWithComputed = {
    id: string;
    source_id: string;
    name: string;
    brand: string | null;
    price: number | null;
    purchase_date: string | null;
    purchase_channel: string | null;
    category: string | null;
    color: string | null;
    image: string | null;
    extra: Record<string, unknown>;
    days_owned: number | null;
    daily_cost: number | null;
};

export type SourceStats = {
    source_id: string;
    source_name: string;
    item_count: number;
    total_value: number;
};

export type ItemStats = {
    total_items: number;
    total_value: number;
    average_daily_cost: number;
    by_source: SourceStats[];
};

export type ItemData = {
    sources: ItemSourceInfo[];
    items: ItemWithComputed[];
    stats: ItemStats;
};

export type ItemSortKey = 'name' | 'price' | 'daily_cost' | 'date' | 'days_owned';
export type ItemSortOrder = 'asc' | 'desc';

File: src-tauri/src/models/item.rs
use serde::{Deserialize, Serialize};
use std::collections::HashMap;

// ── Config file ──

#[derive(Deserialize)]
pub struct ItemSourceFile {
    pub version: u32,
    pub sources: Vec<ItemSource>,
}

#[derive(Deserialize)]
pub struct ItemSource {
    pub id: String,
    pub name: String,
    pub path: String,
}

// ── Response structures ──

#[derive(Serialize)]
pub struct ItemSourceInfo {
    pub id: String,
    pub name: String,
    pub item_count: usize,
}

#[derive(Serialize)]
pub struct ItemWithComputed {
    pub id: String,
    pub source_id: String,
    pub name: String,
    pub brand: Option<String>,
    pub price: Option<f64>,
    pub purchase_date: Option<String>,
    pub purchase_channel: Option<String>,
    pub category: Option<String>,
    pub color: Option<String>,
    pub image: Option<String>,
    pub extra: HashMap<String, serde_json::Value>,
    pub days_owned: Option<u64>,
    pub daily_cost: Option<f64>,
}

#[derive(Serialize)]
pub struct SourceStats {
    pub source_id: String,
    pub source_name: String,
    pub item_count: usize,
    pub total_value: f64,
}

#[derive(Serialize)]
pub struct CategoryStats {
    pub name: String,
    pub item_count: usize,
    pub total_value: f64,
}

#[derive(Serialize)]
pub struct ItemStats {
    pub total_items: usize,
    pub total_value: f64,
    pub average_daily_cost: f64,
    pub by_source: Vec<SourceStats>,
    pub by_category: Vec<CategoryStats>,
}

#[derive(Serialize)]
pub struct ItemData {
    pub sources: Vec<ItemSourceInfo>,
    pub items: Vec<ItemWithComputed>,
    pub stats: ItemStats,
}
